src/app/api/controllers/userController.js

```diff
@@ -0,0 +1,97 @@
+import express from "express";
+import userService from "../services/userService.js";
+import crypto from "crypto";
+
+let tokenStorage = {};
+function makeToken() {
+  return crypto.randomBytes(32).toString("hex");
+}
+
+let cookieOptions = {
+  httpOnly: true,
+  secure: true,
+  sameSite: "strict",
+};
+
+const getUserById = async (req, res) => {
+  const { id } = req.params;
+  res.send(`User with ID: ${id} got`);
+};
+
+const loginUser = async (req, res) => {
+  let { username, password } = req.body;
+  if (!username || !password) {
+    return res.status(400).send("Missing features");
+  }
+
+  try {
+    let result = await userService.loginUser(username, password);
+    if (result) {
+      for (let [key, val] of Object.entries(tokenStorage)) {
+        if (val === username.toUpperCase()) {
+          console.log("Related token", key);
+          return res.cookie("token", key, cookieOptions).send();
+        }
+      }
+
+      let token = makeToken();
+      console.log("Generated token", token);
+      tokenStorage[token] = username.toUpperCase();
+      return res.cookie("token", token, cookieOptions).send();
+    }
+    return res.status(400).send("Incorrect password");
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const createUser = async (req, res) => {
+  const { username, email, cemail, password, cpassword } = req.body;
+
+  if (!username || !email || !cemail || !password || !cpassword) {
+    return res.status(400).send("Missing required fields");
+  }
+
+  if (email.toLowerCase() !== cemail.toLowerCase()) {
+    return res.status(400).send("Email does not match");
+  }
+
+  if (password !== cpassword) {
+    return res.status(400).send("Password does not match");
+  }
+
+  try {
+    await userService.createUser(username, email, password);
+    let token = makeToken();
+    console.log("Generated token", token);
+    tokenStorage[token] = username.toUpperCase();
+    return res.cookie("token", token, cookieOptions).send();
```

Comment:
It'd probably be better to only create the user here and return 201 with a redirect to the login API route, so as to preserve the function "only" creating the user and the login function being the "only" thing that can log you in.

If it's too hard to do quickly, I'm not that worried about it, though.